modules/javafx.media/src/main/native/jfxmedia/platform/osx/avf/AVFAudioSpectrumUnit.cpp

```diff
@@ -192,8 +191,8 @@ void AVFAudioSpectrumUnit::UpdateBands(int size, const float* magnitudes, const
     // Call our listener to dispatch the spectrum event
     if (mSpectrumCallbackProc) {
         double duration = (double) mSamplesPerInterval / (double) 44100;
-        double timestamp = mFirstBufferDelivered ? -1.0 : 0.0;
-        mSpectrumCallbackProc(mSpectrumCallbackContext, duration, timestamp);
+        // Timestamp is ignored and it will be queried from event loop
```

Comment:
You might also note that the reason it is ignored is because it will deadlock if you read it from the media thread.